src/core/Database.cpp

```diff
@@ -355,3 +320,34 @@ const CompositeKey & Database::key() const
     return m_data.key;
 }
 
+QVariantMap& Database::kdfParams() {
+    return m_data.kdfParams;
+}
+
+void Database::setKdfParams(QVariantMap params) {
```

Comment:
I don't disagree that a const reference is better here.

> BUT: as a user of the class you don't know that. 

You should know that, because that is what the Qt documentation says:

> A shared class consists of a pointer to a shared data block that contains a reference count and the data.

Let's not talk about if the documentation was incorrect, because that's another problem.